drivers/console/ipm_console_receiver.c

```diff
@@ -66,24 +66,12 @@ static void ipm_console_thread(void *arg1, void *arg2, void *arg3)
 			++pos;
 		}
 
-		/* ISR may have disabled the channel due to full buffer at
-		 * some point. If that happened and there is now room,
-		 * re-enable it.
-		 *
-		 * Lock interrupts to avoid pathological scenario where
-		 * the buffer fills up in between enabling the channel and
-		 * clearing the channel_disabled flag.
-		 */
-		if (driver_data->channel_disabled &&
-		    ring_buf_space_get(&driver_data->rb)) {
-			key = irq_lock();
-			ipm_set_enabled(driver_data->ipm_device, 1);
-			driver_data->channel_disabled = 0;
-			irq_unlock(key);
-		}
 	}
 }
 
+#ifdef CONFIG_SMP
+__attribute__((optimize("O0")))
```

Comment:
Ouch. Why do we have to do this?

Reply:
without this, compiler will do some optimization for “while (ring_buf_space_get(&driver_data->rb) == 0）{}； ”  and it only call "ring_buf_space_get(&driver_data->rb) " once. And it broken monitoring ring buffer's space.

Reply:
If that's true, then the problem is a missing "volatile" somewhere underneath the inlined implementation of ring_buf_space_get().  We should only be changing optimizer flags for known optimizer bugs.

Reply:
yes, will add volatile for ring_buffer's head tail and size.